### PR TITLE
Whisker

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
This PR includes all the commits in #360.

There are two lines that are commented out with browser() calls at them that still need to be tested. 

More to come, but if anyone has time to look this over, I'm looking for a review or two.

### CODE REVIEW

File: R/initializeProcessInputs.R [hunk 1]
```diff
@@ -18,30 +19,37 @@ defaultProcessInputs <- function(algorithm, wps_url, wps_version){
   checkException(doc)
   
   # -- get optional arguments --
-  optionNd	<-	getNodeSet(doc,'//DataInputs/Input[@minOccurs=0]/ows:Identifier')
+  optionNd	<-	xml2::xml_find_all(doc,'//DataInputs/Input[@minOccurs=0]/ows:Identifier')
   optionLs	<-	vector("list",length(optionNd))
   optionLs[]	<-	NA # "NA" is the equivalent of "optional" as an input
-  names(optionLs)	<-	sapply(optionNd,xmlValue)
+  names(optionLs)	<-	sapply(optionNd,xml2::xml_text)
```
Comment (wdwatkins) [inline R/initializeProcessInputs.R]: I think `xml_text` accepts a nodeset, so you wouldn't  need to `sapply`

File: R/initializeProcessInputs.R [hunk 2]
```diff
@@ -18,30 +19,37 @@ defaultProcessInputs <- function(algorithm, wps_url, wps_version){
...
   
   # -- get and set default values --
   defaultTag <- '//DataInputs/Input/LiteralData/DefaultValue'
-  defaultNd	<-	getNodeSet(doc,paste0(defaultTag, '/parent::node()[1]/DefaultValue'))
-  defaultLs	<-	vector("list",length(sapply(defaultNd,xmlValue)))
-  defaultLs[]	<-	sapply(defaultNd,xmlValue)
-  names(defaultLs)	<-	sapply(getNodeSet(doc,paste0(defaultTag, '/parent::node()[1]/preceding-sibling::node()[3]')),xmlValue)	
+  defaultNd	<-	xml2::xml_find_all(doc,paste0(defaultTag, '/parent::node()[1]/DefaultValue'))
+  defaultLs	<-	vector("list",length(sapply(defaultNd,xml2::xml_text)))
+  defaultLs[]	<-	sapply(defaultNd,xml2::xml_text)
```
Comment (wdwatkins) [inline R/initializeProcessInputs.R]: same here, and a few others

File: R/processInputsToXML.R [hunk 3]
```diff
@@ -21,128 +21,121 @@ setGeneric(name="XML",def=function(stencil, fabric, knife){standardGeneric("XML"
...
+  
+  whisker_list["identifier"] <- knife@algorithm[[1]]
+  
+  input_list <- list() # unnamed list for {{#inputs}}
   
   for (i in 1:length(knife@processInputs)){
-    postNm	<-	names(knife@processInputs[i])
-    postVl	<-	knife@processInputs[postNm]
-    if (!is.na(postVl)){
+    input_identifier	<-	names(knife@processInputs[i])
```
Comment (wdwatkins) [inline R/processInputsToXML.R]: bit of goofy whitespace here

File: R/processInputsToXML.R [hunk 4]
```diff
@@ -21,128 +21,121 @@ setGeneric(name="XML",def=function(stencil, fabric, knife){standardGeneric("XML"
...
-  addChildren(queryEL,propNmEL)
-  propNmEL<-	newXMLNode('wfs:PropertyName',newXMLTextNode(stencil@attribute))
-  addChildren(queryEL,propNmEL)
+  # This is the WFS layer/typename
+  geom_list["wfs_typename"] <- as.character(stencil@geom)
+  
+  # This is what the WFS layer calls its geometry property
+  geom_list["wfs_geom_property"] <- "the_geom" # this is only valid for geoserver
+  
+  # This is the attribute property that the GDP will use to label output
```
Comment (wdwatkins) [inline R/processInputsToXML.R]: 👍  for commenting

File: inst/templates/execute_template.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
+<wps:Execute xmlns:wps="{{wps}}" xmlns:ows="{{ows}}" xmlns:ogc="{{ogc}}" xmlns:xlink="{{xlink}}" xmlns:xsi="{{xsi}}" service="{{service}}" version="{{version}}" xsi:schemaLocation="{{schema_location}}">
```
Comment (wdwatkins) [inline inst/templates/execute_template.xml]: 👍

File: R/initializeProcessInputs.R (hunk 2 above)
Comment (dblodgett-usgs) [reply R/initializeProcessInputs.R]: hmmm... ok I just left the sapply and things worked so I didn't worry about it. Will try and switch to straight xml_text call.

File: R/processInputsToXML.R (hunk 3 above)
Comment (dblodgett-usgs) [reply R/processInputsToXML.R]: Yeah... this file has all kinds of tabs in it. Strange.

File: R/initializeProcessInputs.R (hunk 2 above)
Comment (wdwatkins) [reply R/initializeProcessInputs.R]: Yeah, it probably still works fine with sapply

Comment (dblodgett-usgs) [reply R/initializeProcessInputs.R]: I switch this and tests don't work... OK if I leave it?

Comment (wdwatkins) [reply R/initializeProcessInputs.R]: Hmm ok.  Yeah, fine for now at least.

Comment (dblodgett-usgs) [reply R/initializeProcessInputs.R]: Actually, I take that back. I can eliminate a lot of the sapply calls. Will be in my latest testing PR.

**Pull request merged**